### ISSUE TITLE
setup.bash and other symlinks not added first catkin build

### ISSUE THREAD

**Issue by davetcoleman** [catkin/catkin_tools#493]
### System Info

* Operating System: 4.10.0-38-generic #42~16.04.1-Ubuntu
* Python Version: Python 2.7.12
* Version of catkin_tools: catkin_tools 0.4.4
* ROS Distro: Kinetic

### Build / Run Issue

* [x] Works with `catkin_make`
* [ ] Works with `catkin_make_isolated --merge`
* [x] Works with `catkin build`
* [ ] Works with `catkin build -p1`

### Expected Behavior

The first time I build a catkin workspace (``catkin build``) it does not create the symlinks in the ``devel`` folder necessary to actually use the worksapce e.g. setup.bash. This happens even though everything builds successfully with no warnings. The second time I run ``catkin build`` the setup.bash does work just fine. 

### Steps to Reproduce the Issue

Don't use a ``/src`` folder but rather keep the packages in the root of the catkin workspace. Set the source-space as shown in the ``catkin config`` below:

```bash
cd WORKSPACE
rm -rf build devel .catkin_tools logs
catkin config --source-space . --extend /opt/ros/kinetic --cmake-args -DCMAKE_BUILD_TYPE=Release
catkin build
```

The contents of devel will *sometimes* not have ``setup.bash`` but only:

```
.built_by  .catkin  etc  include  lib  .private  share
```

It appears every other time you run ``catkin config`` then ``catkin build``, the ``setup.bash`` file appears and disappears.

The problem goes away when using a proper ``src`` folder

A work-around is to use ``catkin build --no-deps PACKAGE1 PACKAGE2 PACKAGE3`` for each package you want to build explicitly.

**Comment by nalt** [comment]
I ran into the same problem.

**Comment by timonegk** [comment]
Closed in #650.
